### PR TITLE
Fix provider mock

### PR DESCRIPTION
**Pull request by forgetso**
- [x] Fixes for out of date provider mock
- [x] Build and test provider mock on PR
- [x] Build and push provider mock on publish for AMD64
- [x] Adds option to pass gas increase factor when dry running a contract method
- [x] Uses an SVG for the provider GUI header logo

### CODE REVIEW

File: packages/contract/src/contract/helpers.ts [hunk 1]
```diff
@@ -161,9 +161,10 @@ export function getOptions(
     value?: number | BN,
     gasLimit?: WeightV2,
     storageDeposit?: StorageDeposit,
-    increaseGas?: boolean
+    increaseGas?: boolean,
+    gasIncreaseFactor?: number
 ): ContractOptions {
-    const gasIncreaseFactor = increaseGas ? GAS_INCREASE_FACTOR : 1
+    gasIncreaseFactor = increaseGas ? (gasIncreaseFactor ? gasIncreaseFactor : GAS_INCREASE_FACTOR) : 1
```
Comment (HughParry) [inline packages/contract/src/contract/helpers.ts]: `gasIncreaseFactor = increaseGas ? gasIncreaseFactor || GAS_INCREASE_FACTOR : 1`

Comment (forgetso) [reply packages/contract/src/contract/helpers.ts]: You can suggest this as a commit and get it resolved within github. Not sure how you do it though.

File: packages/types/src/provider/api.ts [hunk 2]
```diff
@@ -128,9 +128,9 @@ export interface VerificationResponse {
 }
 
 export interface ImageVerificationResponse extends VerificationResponse {
-    [ApiParams.commitmentId]: Hash
+    [ApiParams.commitmentId]?: Hash
     // The block at which the captcha was requested
-    [ApiParams.blockNumber]: number
+    [ApiParams.blockNumber]?: number
```
Comment (HughParry) [inline packages/types/src/provider/api.ts:133]: Why would the block at which the captcha was requested not be included in the image verification response?

File: packages/contract/src/contract/helpers.ts (hunk 1 above)
Comment (HughParry) [inline packages/contract/src/contract/helpers.ts]:
```suggestion
    gasIncreaseFactor = increaseGas ? (gasIncreaseFactor ? gasIncreaseFactor : GAS_INCREASE_FACTOR) : 1
```

Comment (HughParry) [inline packages/contract/src/contract/helpers.ts]:
```suggestion
gasIncreaseFactor = increaseGas ? gasIncreaseFactor || GAS_INCREASE_FACTOR : 1
```

**Pull request merged**